test/benchmark_utils/test_benchmark_utils.py

```diff
@@ -0,0 +1,829 @@
+import json
+import os
+import re
+import textwrap
+from typing import List, Tuple
+import unittest
+import torch
+import torch.utils.benchmark as benchmark_utils
+from torch.testing._internal.common_utils import IS_WINDOWS, slowTest
+import numpy as np
+
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+
+def load_test_example() -> Tuple[benchmark_utils.CallgrindStats, benchmark_utils.CallgrindStats]:
+    """Hermetic artifact to unit test Callgrind wrapper.
+
+    In addition to collecting counts, this wrapper provides some facilities for
+    manipulating and displaying the collected counts. The results of several
+    measurements are stored in test_callgrind_artifacts.json.
+
+    While FunctionCounts and CallgrindStats are pickleable, the artifacts for
+    testing are stored in raw string form for easier inspection and to avoid
+    baking any implementation details into the artifact itself.
+
+    The json was generated with the following snippet:
+    ```
+        import json
+        import os
+
+        import torch.utils.benchmark as benchmark_utils
+
+        stats_no_data = benchmark_utils.Timer(
+            "y = torch.ones(())"
+        ).collect_callgrind(number=1000)
+
+        stats_with_data = benchmark_utils.Timer(
+            "y = torch.ones((1,))"
+        ).collect_callgrind(number=1000)
+
+        user = os.getenv("USER")
+        def to_entry(fn_counts):
+            return [f"{c} {fn.replace(f'/{user}/', '/test_user/')}" for c, fn in fn_counts]
+
+        artifacts = {
+            "baseline_inclusive": to_entry(stats_no_data.baseline_inclusive_stats),
+            "baseline_exclusive": to_entry(stats_no_data.baseline_exclusive_stats),
+            "ones_no_data_inclusive": to_entry(stats_no_data.stmt_inclusive_stats),
+            "ones_no_data_exclusive": to_entry(stats_no_data.stmt_exclusive_stats),
+            "ones_with_data_inclusive": to_entry(stats_with_data.stmt_inclusive_stats),
+            "ones_with_data_exclusive": to_entry(stats_with_data.stmt_exclusive_stats),
+        }
+
+        with open("/tmp/test_callgrind_artifacts.json", "wt") as f:
+            json.dump(artifacts, f, indent=4)
```

Comment:
The fixture for generating the test data should be actual code and be executed in CI (with some basic sanity test on the output) to prevent it from bitrotting.

Reply:
It's now a proper function, though unlike `expecttest` it doesn't auto-regen since the diff can be quite large due to changed build dir prefixes.